ludwig/utils/config_utils.py

```diff
@@ -90,3 +103,42 @@ def has_pretrained_encoder(config: ModelConfig) -> bool:
         if feature.encoder.is_pretrained():
             return True
     return False
+
+
+def config_uses_llm(config: Union[Dict[str, Any], ModelConfig]) -> bool:
+    """Determine if a config uses an LLM.
+
+    Args:
+        config: Ludwig config object or dictionary
+
+    Returns:
+        True if the model type is LLM or if the model uses and LLM encoder, otherwise False.
+    """
+    uses_llm = False
+
+    # For a valid config, model_type LLM is automatically True
+    # ECD or GBM models need to be checked for at least one LLM text encoder
+    if isinstance(config, ModelConfig):
+        if config.model_type == MODEL_LLM:
+            uses_llm = True
+        else:
+            for feature in config.input_features:
+                if feature.encoder and feature.encoder.type == MODEL_LLM:
+                    uses_llm = True
+                    break
+    elif isinstance(config, dict) and config:
+        if config.get(MODEL_TYPE, MODEL_ECD) == MODEL_LLM:
+            uses_llm = True
+        elif INPUT_FEATURES in config:
+            for feature in config.get(INPUT_FEATURES, []):
+                if feature.get(ENCODER, {}).get(TYPE) == MODEL_LLM:
+                    uses_llm = True
+                    break
+        else:
+            raise ValueError(
+                "Invalid config cannot be checked for LLM usage because it has no input features." f"Config: {config}"
+            )
```

Comment:
Do we think we need these ValueErrors here? I'd imagine that if the config has made it past config validation, then it is a valid config that should have both `input_features` and `output_features` in the dictionary or config object. What do you think?

Reply:
This is specifically for the dict path because there isn't any validation on dicts; config objects aren't subject to the same check. We could also just load dicts into `ModelConfig` here and run the validations, but that seems like an expensive operation just to check for the existence of LLM(s) in a config.

Reply:
That makes sense! I guess my followup question is, when do we envision this utility function being called in Ludwig code?